chain/client-primitives/src/types.rs

```diff
@@ -464,8 +464,8 @@ impl From<near_chain_primitives::Error> for GetGasPriceError {
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct NetworkInfoResponse {
-    pub active_peers: Vec<PeerInfo>,
-    pub num_active_peers: usize,
+    pub connected_peers: Vec<PeerInfo>,
+    pub num_connected_peers: usize,
```

Comment:
This will break JSON RPC `network` backward compatibility since the old names won't be there:

```
$ http post https://rpc.testnet.near.org jsonrpc=2.0 method=network_info params:='[]' id=dontcare
```

It is suboptimal to expect this type to be serializable straight to JSON RPC response, so I suggest we go and fix it in https://github.com/near/nearcore/blob/9e6c2078dae67938a4fa29a13bb502b04819b7b8/chain/jsonrpc-primitives/src/types/network_info.rs#L5-L6, so instead of flattening `NetworkInfoResponse` there, let's have a copy of the old fields over:

```rust
    pub active_peers: Vec<PeerInfo>,
    pub num_active_peers: usize,
    pub peer_max_count: u32,
    pub sent_bytes_per_sec: u64,
    pub received_bytes_per_sec: u64,
    /// Accounts of known block and chunk producers from routing table.
    pub known_producers: Vec<KnownProducer>,
```

 and implement the `From`:

https://github.com/near/nearcore/blob/9e6c2078dae67938a4fa29a13bb502b04819b7b8/chain/jsonrpc-primitives/src/types/network_info.rs#L16-L20

Reply:
That's a very good catch.. And a gap in our testing (as we should have a test that loads the JSON requests from a file - to simulate backwards compatiblity).

Reply:
We should do the code separation. I'm changing this PR to `draft` until we merge https://github.com/near/nearcore/pull/5515

I created a separate issue for `jsonrpc`  https://github.com/near/nearcore/issues/5516